libs/cardano-ledger-api/src/Cardano/Ledger/Api/State/Query.hs

```diff
@@ -185,6 +190,41 @@ queryDRepStakeDistr nes creds
   where
     distr = psDRepDistr . fst $ finishedPulserState nes
 
+-- | Query the stake distribution of the registered DReps. This does not
+-- include the @AlwaysAbstain@ and @NoConfidence@ DReps.
+queryRegisteredDRepStakeDistr ::
+  ConwayEraGov era =>
+  NewEpochState era ->
+  -- | Specify DRep Ids whose stake distribution should be returned. When this set is
+  -- empty, distributions for all of the registered DReps will be returned.
+  Set (Credential 'DRepRole (EraCrypto era)) ->
+  Map (Credential 'DRepRole (EraCrypto era)) Coin
+queryRegisteredDRepStakeDistr nes creds
+  | null creds = distr
+  | otherwise = distr `Map.restrictKeys` creds
+  where
+    distr = Map.map fromCompact $ Map.foldlWithKey' computeDistr mempty registeredDReps
+    registeredDReps = nes ^. nesEsL . esLStateL . lsCertStateL . certVStateL . vsDRepsL
```

Comment:
Indeed! That's what I was doing initially: https://github.com/IntersectMBO/cardano-ledger/commit/026a2cdbcee7b27ac99f3168030bfcf809d1e073#diff-2d5e051866ea7ca159f3c83cb39fa89c4861f702481e40097213311206c73f0cR202
I started to change things up because of the potential `AlwaysAbstain` and `NoConfidence` DRep inclusions, which fortunately won't happen after all. Cheers!